### ISSUE TITLE
Refactor: add custom base distribution class

### ISSUE THREAD

**Issue by sgreenbury** [alan-turing-institute/autoemulate#436]
Following discussion as part of #345 / #411, having an additional base class for distributions would be a useful feature.

It may include the following:
- Subclass torch.distribution
- Having a means to do Transformations (like [TransformedDistribution](https://pytorch.org/docs/stable/distributions.html#torch.distributions.transformed_distribution.TransformedDistribution), or TFP [bijector](https://www.tensorflow.org/probability/api_docs/python/tfp/bijectors/Bijector)), including normalization

See also approach in pyro: https://docs.pyro.ai/en/stable/distributions.html

Relates also to #327.

**Comment by sgreenbury** [comment]
Adding note that this also relates to:
- #430 
- #348 
- #376

**Comment by radka-j** [comment]
@sgreenbury can we close this issue given all functionality in`TransformedEmulator`?

**Comment by sgreenbury** [comment]
Yes I think this makes sense to close as not planned given the current implementations in #474, #551, #726
